geostructures/time.py

```diff
@@ -129,3 +138,35 @@ def issuperset(self, other: TimeInterval) -> bool:
     def union(self, other: TimeInterval) -> TimeInterval:
         """Returns a TimeInterval that spans both time intervals"""
         return TimeInterval(min(self.start, other.start), max(self.end, other.end))
+
+    @classmethod
+    def _get_timeformat(
+        cls,
+        time_str: str,
+        formats: List[str] = _DATE_FORMATS
+    ) -> str:
+        for fmt in formats:
+            try:
+                datetime.strptime(time_str, fmt)
+                return fmt
+            except ValueError:
+                continue
+        raise ValueError(f'Date formate was not recognized; {time_str}')
+
+    @classmethod
+    def from_str(
+        cls,
+        start: str,
+        end: Optional[str] = None,
+        time_format: Optional[Union[str, List[str]]] = None
```

Comment:
I think we should simplify this to `Optional[str]` instead of also allowing a list - it would simplify the code a lot, and what's really the chance that start and end dates are formatted differently? (Also, you're actually only checking the start format anyways)

Reply:
In my test case for this function I did have two formats in the column, so it can happen. This data came from a source that does their own aggregation of data before it is returned to the user. We also check the end date if it is provided.